### PR TITLE
Chore: Added type hints to searches/binary_search.py

### PR DESCRIPTION
**Pull request by UmairKamran**
### **Describe your change:**

Added missing type hints as described in this issue -> #2128 

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by hemanth-kotagiri — APPROVED** [review]
LGTM.
@Panquesito7, I approve of these changes. Please look into it.

**Comment by Panquesito7** [comment]
> Please look into it.

Sorry, I'm not a Python developer/maintainer.
@cclauss.

**Comment by UmairKamran** [comment]
> LGTM.
> @Panquesito7, I approve of these changes. Please look into it.

Hey @hemanth-kotagiri, how can we get this merged? It's becoming stale

**Comment by hemanth-kotagiri** [comment]
@UmairKamran, I do not have the rights for this repository yet to merge your PR. What I'd suggest you is the email or ping @cclauss for the same. 

Thanks!

**Comment by dhruvmanila** [comment]
Closing and reopening this pull request to run checks done by the bot.

File: searches/binary_search.py [hunk 1]
```diff
@@ -219,7 +228,7 @@ def binary_search(sorted_collection, item):
     return None
 
 
-def binary_search_std_lib(sorted_collection, item):
+def binary_search_std_lib(sorted_collection, item) -> Optional[int]:
```
Comment (dhruvmanila) [inline searches/binary_search.py:231]: I think you missed a few in here :)

**Review by dhruvmanila — CHANGES_REQUESTED** [review]
Sorry for the delay. Small changes and this is good to go!

**Comment by cclauss** [comment]
I will fix it and land it in the next few hours.  Instead of parameters that can be either int or None, it would be better if they were only int where we use -1 as the signal value rather than None.

**Comment by dhruvmanila** [comment]
Ah, so you want to use the Sentinel pattern? There's a good article about it: https://python-patterns.guide/python/sentinel-object/

Any reason why you are preferring for an `int` instead of `None`? `None` is a good indication for the absence of value unless `None` is actually a valid value for the given parameter which is where the Sentinel object pattern is used.

**Review by cclauss — APPROVED** [review]

File: searches/binary_search.py (hunk 1 above)
Comment (UmairKamran) [reply searches/binary_search.py]: Good catch! Looks like @cclauss has taken care of this

**Comment by UmairKamran** [comment]
@dhruvmanila @cclauss Thank you for reviewing  :)

**Comment by cclauss** [comment]
> Any reason why you are preferring for an int instead of None?

The expected datatype is int so why make things more complicated if we can find a suitable sentinel value of that datatype?  The current project that I am working on overuses `None` throughout the codebase.  This complicates the addition of type checking (with `Optional`) so many contributors just avoid using type checking which is hurting our ability to spot bugs.

**Pull request merged**